server/scripts/provision-tenant.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash -e
+
+# Install dependencies
+sudo yum update -y
+sudo yum install -y nodejs
+sudo yum install -y jq
+sudo yum install -y python3-pip
+sudo yum install -y npm
+sudo npm install -g aws-cdk
+sudo python3 -m pip install --upgrade setuptools
+sudo python3 -m pip install git-remote-codecommit
+
+# Enable nocasematch option
+shopt -s nocasematch
+
+# Clone the serverless reference solution repository
+export CDK_PARAM_CODE_COMMIT_REPOSITORY_NAME="aws-saas-factory-ref-solution-serverless-saas"
+git clone codecommit://$CDK_PARAM_CODE_COMMIT_REPOSITORY_NAME
+cd $CDK_PARAM_CODE_COMMIT_REPOSITORY_NAME/server/cdk
+npm install
+
+export CDK_PARAM_COMMIT_ID=$(git log --format="%H" -n 1)
+
+# Parse tenant details from the input message from step function
+export CDK_PARAM_TENANT_ID=$tenantId
+export TIER=$tier
+export TENANT_ADMIN_EMAIL=$email
+
+# Define variables
+TENANT_ADMIN_USERNAME="tenant-admin-$CDK_PARAM_TENANT_ID"
+STACK_NAME="serverless-saas-ref-arch-tenant-template-pooled"
+USER_POOL_OUTPUT_PARAM_NAME="TenantUserpoolId"
+API_GATEWAY_URL_OUTPUT_PARAM_NAME="ApiGatewayUrl"
+APP_CLIENT_ID_OUTPUT_PARAM_NAME="UserPoolClientId"
+
+# Deploy the tenant template for platinum tier(silo)
+if [[ $TIER == "PLATINUM" ]]; then
```

Comment:
is it possible to just leverage the step function to do this? Otherwise the create stack in the step function seems redundant. So step function is only to update the stacmk now? It might be ok, but lets discuss this